File: Examples/CSharp/AsposePDF/Bookmarks/UpdateChildBookmarks.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.Bookmarks
{
    public class UpdateChildBookmarks
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Bookmarks();

            //open document
            Document pdfDocument = new Document(dataDir + "UpdateChildBookmarks.pdf");

            //get a bookmark object
            OutlineItemCollection pdfOutline = pdfDocument.Outlines[1];
            
            //get child bookmark object
            OutlineItemCollection childOutline = pdfOutline[1];
            childOutline.Title = "Updated Outline";
            childOutline.Italic = true;
            childOutline.Bold = true;
                        
            //save output
            pdfDocument.Save(dataDir + "UpdateChildBookmarks_out.pdf");

        }
    }
}

File: Examples/VisualBasic/AsposePDF/Bookmarks/UpdateChildBookmarks.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf

Namespace AsposePDF.Bookmarks
    Public Class UpdateChildBookmarks
        Public Shared Sub Run()
            ' ExStart:UpdateChildBookmarks
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Bookmarks()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("UpdateChildBookmarks.pdf"))

            ' Get a bookmark object
            Dim pdfOutline As OutlineItemCollection = pdfDocument.Outlines(1)

            ' Get child bookmark object
            Dim childOutline As OutlineItemCollection = pdfOutline(1)
            childOutline.Title = "Updated Outline"
            childOutline.Italic = True
            childOutline.Bold = True
            dataDir = dataDir & Convert.ToString("UpdateChildBookmarks_out_.pdf")
            ' Save output
            pdfDocument.Save(dataDir)
            ' ExEnd:UpdateChildBookmarks
            Console.WriteLine(Convert.ToString(vbLf & "Child bookmarks updated successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace